include/swift-c/DependencyScan/DependencyScan.h

```diff
@@ -0,0 +1,339 @@
+#define SWIFT_C_DEPENDENCY_SCAN_H
+
+#include "DependencyScanMacros.h"
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/// The version constants for the SwiftDependencyScan C API.
+/// SWIFTSCAN_VERSION_MINOR should increase when there are API additions.
+/// SWIFTSCAN_VERSION_MAJOR is intended for "major" source/ABI breaking changes.
+#define SWIFTSCAN_VERSION_MAJOR 0
+#define SWIFTSCAN_VERSION_MINOR 1
+
+SWIFTSCAN_BEGIN_DECLS
+
+//=== Public Scanner Data Types -------------------------------------------===//
+
+/**
+ * A character string used to pass around dependency scan result metadata.
+ * Lifetime of the string is strictly tied to the object whose field it
+ * represents. When the owning object is released, string memory is freed. Use
+ * \c swiftscan_get_C_string() to retrieve the string data.
+ */
+typedef struct {
+  const void *data;
+  size_t length;
+} swiftscan_string_ref_t;
+
+typedef struct {
+  swiftscan_string_ref_t *strings;
+  size_t count;
+} swiftscan_string_set_t;
+
+typedef enum {
+  SWIFTSCAN_DEPENDENCY_INFO_SWIFT_TEXTUAL = 0,
+  SWIFTSCAN_DEPENDENCY_INFO_SWIFT_BINARY = 1,
+  SWIFTSCAN_DEPENDENCY_INFO_SWIFT_PLACEHOLDER = 2,
+  SWIFTSCAN_DEPENDENCY_INFO_CLANG = 3
+} swiftscan_dependency_info_kind_t;
+
+/// Opaque container of the details specific to a given module dependency.
+typedef struct swiftscan_module_details_s *swiftscan_module_details_t;
+
+/// Opaque container to a dependency info of a given module.
+typedef struct swiftscan_dependency_info_s *swiftscan_dependency_info_t;
+
+/// Opaque container to an overall result of a dependency scan.
+typedef struct swiftscan_dependency_graph_s *swiftscan_dependency_graph_t;
+
+/// Opaque container to contain the result of a dependency prescan.
+typedef struct swiftscan_import_set_s *swiftscan_import_set_t;
+
+/// Full Dependency Graph (Result)
+typedef struct {
+  swiftscan_dependency_info_t *modules;
+  size_t count;
+} swiftscan_dependency_set_t;
+
+//=== Batch Scan Input Specification --------------------------------------===//
+
+/// Opaque container to a container of batch scan entry information.
+typedef struct swiftscan_batch_scan_entry_s *swiftscan_batch_scan_entry_t;
+
+typedef struct {
+  swiftscan_batch_scan_entry_t *modules;
+  size_t count;
+} swiftscan_batch_scan_input_t;
+
+typedef struct {
+  swiftscan_dependency_graph_t *results;
+  size_t count;
+} swiftscan_batch_scan_result_t;
+
+//=== Scanner Invocation Specification ------------------------------------===//
+
+/// Opaque container of all relevant context required to launch a dependency
+/// scan (command line arguments, working directory, etc.)
+typedef struct swiftscan_scan_invocation_s *swiftscan_scan_invocation_t;
+
+//=== String Functions ----------------------------------------------------===//
+
+/// Retrieve the character data associated with the given string.
+SWIFTSCAN_PUBLIC const char *
+swiftscan_get_C_string(swiftscan_string_ref_t string);
```

Comment:
Is it guaranteed that a `swiftscan_string_ref_t` will always point to a null-terminated buffer?
If not, I'd suggest to avoid having this function and on the Swift side create a Swift string using pointer+size.
It will be safer since you won't have to worry about always providing null-terminated strings.

Reply:
It is not guaranteed, indeed.
Removed the function and will instead build a client-side utility as you describe. 
Thanks for the suggestion!